tests/testthat/test-3agglomerate.R

```diff
@@ -145,7 +145,7 @@ test_that("agglomerate", {
     # Test that dimentionality is the same when NA values are removed.
     expect_equal(length((unique(rowData(tse)[,"Family"]))[ !is.na(unique(rowData(tse)[,"Family"])) ]),
                  nrow(agglomerateByRank(tse, rank="Family", na.rm = TRUE)))
-      expect_equal(length((unique(rowData(tse)[,"Family"]))[ !is.na(unique(rowData(tse)[,"Family"])) ]),
+    expect_equal(length((unique(rowData(tse)[,"Family"]))[ !is.na(unique(rowData(tse)[,"Family"])) ]),
                  nrow(agglomerateByVariable(tse, f="Family", MARGIN = 1, agg.na.rm = TRUE)))
```

Comment:
Can you extract these values first and then check that they match? Also, agg.na.rm is not parameter of agglomerateByVariable, it is parameter of agglomerateByPrevalence.

agglomerateByVariable has parameter na.rm. So I am wondering if these functions work correctly.... Although it might be that there are no NA values in family level. Can you switch using the lowest rank in rowData? That should have NA values.

Reply:
Looks good though. With these tests, good to go (when fixed)

Reply:
You are right. this is what i get: 
```
> data(GlobalPatterns, package="mia")
> tse <- GlobalPatterns
> actual <- unique(rowData(tse)[,"Species"])[ !is.na(unique(rowData(tse)[,"Species"]))]
> actual1 <- agglomerateByRank(tse, rank="Species", na.rm = TRUE)
> actual2 <- agglomerateByVariable(tse, f="Species", MARGIN = 1, na.rm = TRUE)
> expect_equal(length(actual), nrow(actual1))
> expect_equal(length(actual), nrow(actual2))
Error: length(actual) not equal to nrow(actual2).
1/1 mismatches
[1] 944 - 945 == -1
```
na.rm doesn't make a difference in agglomerateByVariable.  I also checked source and it doesn't quite use na.rm.